### DISCUSSION TITLE
DBFirst，Oracle11，US7ASCII字符集，生成Model时报错ora00911

### DISCUSSION THREAD

**Discussion by 454063542** [dotnetcore/FreeSql#1421] [Q&A]
使用命令：FreeSql.Generator -Razor 1 -NameOptions 0,0,0,1 -NameSpace  XX.XXX -DB "Oracle,user id=XXpassword=XX;data source=//192.168.1.XX:1521/orcl;pooling=true;max pool size=2"  进行Model生成时，报错如下
![1676301239020](https://user-images.githubusercontent.com/125287349/218496193-2e5f9444-6ee8-4324-8f09-81257aaa1d7d.png)

**Comment by 454063542** [comment]
应该是由于字符集为US7ASCII导致的，我生成单个表的模型是可以的。但生成的中文注释是乱码。 在使用generator的Model生成命令时，无法指定Oracle的连接字符串为Provider=OraOLEDB.Oracle，会报错
![1676303311950](https://user-images.githubusercontent.com/125287349/218504968-7881ae9b-f8de-463d-a7d2-79299d4f3e9a.png)

**Comment by 2881099** [comment]
尝试步骤：

1. 了解 https://freesql.net/guide/freesql-provider-oracle.html

2. 修改 FreeSql.Generator 源码，将 oracle adonet 改成第一步提到的 oledb 方式

3. 修改版本号编译后，使用 dotnet tool update FreeSql.Generator -g --add-source 本地编译的目录路径
